library/data/registry_entry_attachment/src/main/java/org/quiltmc/qsl/registry/attachment/api/RegistryEntryAttachment.java

```diff
@@ -287,6 +288,26 @@ default Optional<V> get(R entry) {
 	 */
 	boolean remove(TagKey<R> tag);
 
+	/**
+	 * The entry validation can prevent entries from being shown in the attachment.
```

Comment:
"Being shown" is misleading; it implies that this validator will silently drop entries, but you seem to throw a hard exception when an entry that fails the predicate is added:

https://github.com/QuiltMC/quilt-standard-libraries/blob/c8e05fbe16d52e2daf7df8d80743726060f57167/library/data/registry_entry_attachment/src/main/java/org/quiltmc/qsl/registry/attachment/impl/RegistryEntryAttachmentImpl.java#L197-L199